### PR TITLE
RUBY-2578 Repair Ubuntu 14.04 based configs

### PR DESCRIPTION
**Pull request by p-mongo**
Reverts mongodb/mongo-ruby-driver#2320

### CODE REVIEW

**Comment by p-mongo** [comment]
Looks like OCSP verifier tests need a new certificate.

File: .evergreen/config-atlas.yml [hunk 1]
```diff
@@ -606,10 +606,9 @@ axes:
       - id: ubuntu1604
         display_name: "Ubuntu 16.04"
         run_on: ubuntu1604-small
-      # https://jira.mongodb.org/browse/RUBY-2758
-      # - id: ubuntu1404
-      #   display_name: "Ubuntu 14.04"
-      #   run_on: ubuntu1404-small
+      - id: ubuntu1404
```
Comment (comandeo) [inline .evergreen/config-atlas.yml:609]: I am sure I will be confused when id ubuntu1404 while it is actually ubuntu1604. Can we change the id? Or just use existing 1604?

**Review by comandeo — APPROVED** [review]

File: .evergreen/config-atlas.yml (hunk 1 above)
Comment (p-mongo) [reply .evergreen/config-atlas.yml:609]: So just switch everything to ubuntu1604?

Comment (comandeo) [reply .evergreen/config-atlas.yml:609]: As I understand the changes, now both ubuntu1404 and and ubuntu1604 are based on `ubuntu1604-small`. So, they are basically the same, aren't they? If they are the same, I would rather stick with just one. If they are different, I would rename `ubuntu1404` to something that better reflects what it is.

**Review by comandeo — APPROVED** [review]

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/RUBY-2758

**Pull request merged**